### PR TITLE
Solution

### PR DESCRIPTION
**Pull request by RangerLeoo**

### CODE REVIEW

File: app/cinema/hall.py [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+        self.number = number
+
+    def movie_session(
+            self,
+            movie_name: str,
+            customers: list,
+            cleaning_staff: Cleaner
+    ) -> None:
+        print(f'"{movie_name}" started in hall number {self.number}.')
+        for people in customers:
```
Comment (Serhii-Leonenko) [inline app/cinema/hall.py:16]: I think it isn't the correct naming. People is plural, so you can use person or customer:
`for customer in customers`

File: app/cinema/hall.py [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+
+class CinemaHall:
+    def __init__(self, number: int) -> None:
+        self.number = number
+
+    def movie_session(
+            self,
+            movie_name: str,
+            customers: list,
```
Comment (Serhii-Leonenko) [inline app/cinema/hall.py:12]:
```suggestion
            customers: list[Customer],
```

use more descriptive type annotations in all similar cases too

File: app/cinema/hall.py [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+            cleaning_staff: Cleaner
+    ) -> None:
+        print(f'"{movie_name}" started in hall number {self.number}.')
+        for people in customers:
+            Customer.watch_movie(self=people, movie=movie_name)
+        print(f'"{movie_name}" ended.')
+        Cleaner.clean_hall(cleaning_staff, self.number)
```
Comment (Serhii-Leonenko) [inline app/cinema/hall.py:19]:
```suggestion
            customer.watch_movie(movie_name)
        print(f'"{movie_name}" ended.')
        cleaning_staff.clean_hall(self.number)
```

Call the methods like these

File: app/main.py [hunk 4]
```diff
@@ -1,6 +1,30 @@
...
+        movie: str
+) -> None:
+    customers_list = []
+    for people in customers:
+        customers_list.append(Customer(
+            name=people["name"],
+            food=people["food"]
+        ))
+
+    for cust in customers_list:
```
Comment (Serhii-Leonenko) [inline app/main.py:20]: Use more descriptvie variable names

File: app/main.py [hunk 5]
```diff
@@ -1,6 +1,30 @@
...
+        CinemaBar.sell_product(product=cust.food, customer=cust)
```
Comment (Serhii-Leonenko) [inline app/main.py:21]: create only one instance of CinemaBar and use it for every customer and you can do this in the same loop where you create customers.
`bar = CinemaBar()`
`for customer in customer_list:`
    `bar.sell_product(...)`

Comment (Serhii-Leonenko) [reply app/main.py:21]: The same about cinema hall,you have a new_hall instance, call it's method 
`new_hall.movie_session(...)`

**Review by Serhii-Leonenko — CHANGES_REQUESTED** [review]

**Review by andrewbelo — APPROVED** [review]

**Pull request closed without merging**